cmd/prometheus/main.go

```diff
@@ -180,6 +180,9 @@ func (c *flagConfig) setFeatureListOptions(logger log.Logger) error {
 			case "extra-scrape-metrics":
 				c.scrape.ExtraMetrics = true
 				level.Info(logger).Log("msg", "Experimental additional scrape metrics enabled")
+			case "metadata-in-wal":
```

Comment:
Because the use case is not just to store in WAL, but to also remote-write (and maybe more later?).
```suggestion
			case "store-metadata":
```

Reply:
To keep it simple, I think there should be only one flag to enable everything about metadata. And right now we only have the remote-write use-case, right? Maybe `"series-metadata"`?

Reply:
Oh wait, I see your other suggestion. metadata-storage sounds better.

Reply:
I also agree that a single flag is a better choice; I renamed this to `metadata-storage` along the other suggestions.